canary/generator/download/download_whole.py

```diff
@@ -0,0 +1,38 @@
+from collections import defaultdict
+import json
+from json import JSONDecodeError
+from ast import literal_eval
+import sys
+import os
+
+import requests
+
+from .query import query
+
+if __name__ == '__main__':
+
+    dates_versions = requests.get(
+        'https://aggregates.telemetry.mozilla.org/aggregates_by/build_id/channels/nightly/dates/').text
+    dates_versions = literal_eval(dates_versions)
```

Comment:
I don't think you need `literal_eval` here, that is for Python literals, while the service returns JSON data (which just happens to fit the python model).
Instead we can use:
```python
response = requests.get(...)
dates_versions = json.loads(response.text)
```

The same below for other `request()s`.